### ISSUE TITLE
DEV linter errors persist

### ISSUE THREAD

**Issue by tablatronix** [gepd/Deviot#264]
It seems that linter context errors, stay in a buffer even after recompile.
If there is an error, previous errors show back up, even after closing them.

### reproduce

make syntax error
compile bad
close red `error`s

compile successfully
make different syntax error
compile bad with a different error

same errors show back up from first time

**Comment by tablatronix** [comment]
any ideas how to clear these out without restarting?

![screen shot 2018-10-31 at 8 31 53 am 2](https://user-images.githubusercontent.com/807787/47809746-a064ab00-dd0f-11e8-85c6-3a5fe5a18f18.png)

**Comment by tablatronix** [comment]
`self.errs_by_file[file]` is never cleared, errors just keep appending.
not sure where to put a clear for compile command only? or how to get self.file in command, for now I am just emptying the entire thing on command.

It seems not very efficient to be processing errors and phantoms on data constantly, not sure if there is a way to do it once at end, probably doesn't matter seemed like excessive processing.

**Comment by gepd** [comment]
I've pushed a temporal fix for this issue, let me know if it's better

**Comment by tablatronix** [comment]
No, also I did not see any commits related to this fix, you sure it pushed?

**Comment by tablatronix** [comment]
if you mean this?
https://github.com/gepd/Deviot/commit/5560fb3633c7c66f6332ee18ef12530e1e60ef37

thats just the setting toggle

**Comment by tablatronix** [comment]
works great, what do you think about clearing errors after successful compile, if you do not X them they just stick around
